src/ray/object_manager/object_manager_client_pool.h

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "absl/base/thread_annotations.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/strings/str_cat.h"
+#include "absl/synchronization/mutex.h"
+#include "ray/common/id.h"
+#include "ray/gcs/gcs_client.h"
+#include "ray/rpc/object_manager/object_manager_client.h"
+
+using absl::optional;
```

Comment:
nits: [don't use namespace in header](https://docs.microsoft.com/en-us/cpp/cpp/namespaces-cpp?view=msvc-160#:~:text=In%20general%2C%20avoid%20putting%20using,names%20in%20a%20header%20file.)